sw/device/silicon_creator/rom/bootstrap.c

```diff
@@ -179,16 +179,20 @@ static rom_error_t bootstrap_page_program(uint32_t addr, size_t byte_count,
       .write = kMultiBitBool4True,
       .erase = kMultiBitBool4False,
   });
-  // Wrap to the start of the flash programming page if `addr` is not aligned.
+  // Perform two writes if the start address is not page-aligned (256 bytes).
+  // Note: Address is flash-word-aligned (8 bytes) due to the check above.
   rom_error_t err_0 = kErrorOk;
   size_t prog_page_misalignment = addr & kFlashProgPageMask;
   if (prog_page_misalignment > 0) {
     size_t word_count =
         (kFlashProgPageSize - prog_page_misalignment) / sizeof(uint32_t);
+    if (word_count > rem_word_count) {
+      word_count = rem_word_count;
+    }
     err_0 = flash_ctrl_data_write(addr, word_count, data);
     rem_word_count -= word_count;
     data += word_count * sizeof(uint32_t);
-    addr &= ~kFlashProgPageMask;
+    addr += word_count * sizeof(uint32_t);
```

Comment:
This is not correct. A PAGE_PROGRAM command modifies only a single page (hence the name ;) ).

Reply:
Ah man, I was totally confused! Thanks for catching! A comment would be nice here hehe

Reply:
Done